File: examples/verify.py
"""Verify a fresh signed envelope against the public Notary Protocol endpoint."""

import json
import os
from urllib.request import Request, urlopen

base_url = os.environ.get("NOTARY_URL", "https://notary-protocol.notary-labs.workers.dev")
with urlopen(f"{base_url}/v1/demo", timeout=10) as response:
    envelope = json.load(response)

request = Request(
    f"{base_url}/v1/verify",
    data=json.dumps(envelope).encode(),
    headers={"content-type": "application/json"},
    method="POST",
)
with urlopen(request, timeout=10) as response:
    receipt = json.load(response)

print(json.dumps({
    "receiptId": receipt["id"],
    "valid": receipt["valid"],
    "checks": len(receipt["checks"]),
    "violations": receipt["violations"],
}, indent=2))

File: examples/verify.mjs
const baseUrl = process.env.NOTARY_URL ?? "https://notary-protocol.notary-labs.workers.dev";

const envelope = await fetch(`${baseUrl}/v1/demo`).then((response) => response.json());
const response = await fetch(`${baseUrl}/v1/verify`, {
  method: "POST",
  headers: { "content-type": "application/json" },
  body: JSON.stringify(envelope)
});
const receipt = await response.json();

console.log(JSON.stringify({ receiptId: receipt.id, valid: receipt.valid, checks: receipt.checks.length, violations: receipt.violations }, null, 2));
if (!receipt.valid) process.exitCode = 1;
